core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -523,18 +547,26 @@ private[sql] final case class ColumnTableScan(
 
       if (!isWideSchema) {
         moveNextCode.append(genCodeColumnNext(ctx, decoderVar, bufferVar,
-          cursorVar, batchOrdinal, attr.dataType, notNullVar)).append('\n')
+          cursorVar, batchOrdinal, attr.dataType, notNullVar, false)).append('\n')
         val (ev, bufferInit) = genCodeColumnBuffer(ctx, decoderVar, bufferVar,
           cursorVar, attr, notNullVar, weightVarName, false)
         bufferInitCode.append(bufferInit)
         ev
       } else {
+        if (isWideSchema) {
```

Comment:
Too many wide schema checks everywhere is making it quite complex. Need to refactor this post 0.9.